pkg/component/input_spec.go

```diff
@@ -12,13 +12,14 @@ import (
 
 // InputSpec is the specification for an input type.
 type InputSpec struct {
-	Name        string      `config:"name" yaml:"name"  validate:"required"`
-	Aliases     []string    `config:"aliases,omitempty" yaml:"aliases,omitempty"`
-	Description string      `config:"description" yaml:"description" validate:"required"`
-	Platforms   []string    `config:"platforms" yaml:"platforms" validate:"required,min=1"`
-	Outputs     []string    `config:"outputs,omitempty" yaml:"outputs,omitempty"`
-	Shippers    []string    `config:"shippers,omitempty" yaml:"shippers,omitempty"`
-	Runtime     RuntimeSpec `config:"runtime,omitempty" yaml:"runtime,omitempty"`
+	Name            string      `config:"name" yaml:"name"  validate:"required"`
+	Aliases         []string    `config:"aliases,omitempty" yaml:"aliases,omitempty"`
+	Description     string      `config:"description" yaml:"description" validate:"required"`
+	Platforms       []string    `config:"platforms" yaml:"platforms" validate:"required,min=1"`
+	Outputs         []string    `config:"outputs,omitempty" yaml:"outputs,omitempty"`
+	OptionalActions []string    `config:"optional_actions,omitempty" yaml:"optional_actions,omitempty"`
```

Comment:
Why the name `optional_actions`? Doesn't seem to fit what this is really doing. This specification file is critical to how Elastic Agent operates and naming of fields I believe is very important. Once a name is used it cannot change, as we are saying that external components can be plugged into the Elastic Agent using these spec files.

This seems to be more a list of actions that the Elastic Agent receives that the component/unit also wants to receive. 

I need to think more on the naming of it, maybe like `agent_actions` or `action_proxy`. I will think on it, but I don't believe the current name makes sense.

Reply:
:-) It was the `agent_actions` originally. was asked to change.

Reply:
This is a small enough change we can discuss it in a Github issue, or even just keep using this comment thread.

In the original suggestion of `optional_actions`, the thinking was that long term it would act like an action capabilities list. This wouldn't let us distinguish between custom actions and agent actions that need to be forwarded to components like this, but we could probably infer that from the action name.

If we want a clear separation between actions that need to be forwarded or proxied like this and custom actions, then I'm fine with calling this `proxied_actions` in line with Blake's original suggestion of `action_proxy` but reading a bit more like a sentence.